redismodule-rs-macros/src/info_section.rs

```diff
@@ -0,0 +1,109 @@
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use quote::{quote, ToTokens};
+use syn::{parse_macro_input, Data, DataStruct, DeriveInput, Fields};
+
+/// Returns `true` if the `type_string` provided is a type of a map
+/// supported by the [`crate::InfoSection`].
+pub fn is_supported_map(type_string: &str) -> bool {
+    /// A list of supported maps which can be converted to a dictionary for
+    /// the [`redis_module::InfoContext`].
+    const ALL: [&str; 2] = ["BTreeMap", "HashMap"];
```

Comment:
Why do we need this validation, it will raise a compilation error anyway if the type will not implement the from trait no?

Reply:
So this means we can not put another struct as a dictionary section?

Reply:
Feels like we need another derive, `InfoDictionary`...

Reply:
And also an attribute on the field that tells if its a dictionary probably... but I do not think it is mandatory for now so we can save it for future improvements ...

Reply:
> And if we mark it with an attribute, we still won't be sure (it still may fail).
It will failed on compile time, but we will know which code to generate for the field ...